xbmc/pvr/channels/PVRChannelGroup.cpp

```diff
@@ -376,63 +376,59 @@ void CPVRChannelGroup::SortByChannelNumber(void)
 
 CPVRChannelPtr CPVRChannelGroup::GetByClient(int iUniqueChannelId, int iClientID) const
 {
+  CPVRChannelPtr retval;
   CSingleLock lock(m_critSection);
 
-  for (unsigned int ptr = 0; ptr < m_members.size(); ptr++)
+  for (std::vector<PVRChannelGroupMember>::const_iterator it = m_members.begin(); !retval && it != m_members.end(); ++it)
   {
-    PVRChannelGroupMember groupMember = m_members.at(ptr);
-    if (groupMember.channel->UniqueID() == iUniqueChannelId &&
-        groupMember.channel->ClientID() == iClientID)
-      return groupMember.channel;
+    if ((*it).channel->UniqueID() == iUniqueChannelId &&
+        (*it).channel->ClientID() == iClientID)
+      retval = (*it).channel;
```

Comment:
I don't think that it's more complex to not put half of the break condition in the for line and another one in the loop.